resources/views/components/organisms/_o-table-of-contents.blade.php

```diff
@@ -7,6 +7,18 @@
                 $isExpanded = true;
             }
             $isActive = isset($currentArticle) && $item->id === $currentArticle->id;
+
+            if ($item->type === App\Enums\DigitalPublicationArticleType::Grouping) {
+                $href = route(
+                    'collection.publications.digital-publications.showListing',
+                    [
+                        'id' => $item->digitalPublication->id,
+                        'slug' => $item->digitalPublication->getSlug()
+                    ]
+                ) . '#' . Str::kebab($item->title);
+            } else {
+                $href = $item->url;
+            }
```

Comment:
`DigitalPublicationArticle->getUrlAttribute()` makes a call to `DigitalPublicationArticlePresenter->getCanonicalUrl()`. Instead of this logic living in the view, I think you can update the `getCanonicalUrl()` method to return the anchor-linked URL if the article type is `grouping`, so that the view can simply call `$item->url`.